### ISSUE TITLE
[ui5-dynamic-page][SF_ACC]: missing aria-label for headerArea slot

### ISSUE THREAD

**Issue by GongRichard** [UI5/webcomponents#12274]
### Bug Description

There defined role="region" for the headerArea slot of ui5-dynamic-page, but there was no aria-label.

<img width="1216" height="976" alt="Image" src="https://github.com/user-attachments/assets/679af497-1684-4fef-834a-f2b2eec937e4" />

### Affected Component

ui5-dynamic-page

### Expected Behaviour

I think it's better to let the application define the role and aria-label, and we can remove role="region" here.
If it must define role="region" here, it should also define aria-label here. And also, it is better for application to have the capability to customize the aria-label.

### Steps to Reproduce

1. open sample https://sap.github.io/ui5-webcomponents-react/v2/?path=/story/layouts-floorplans-dynamicpage--default
2. check the dom of headerArea slot

### Log Output, Stack Trace or Screenshots

<img width="1216" height="976" alt="Image" src="https://github.com/user-attachments/assets/679af497-1684-4fef-834a-f2b2eec937e4" />

### Priority

Medium

### UI5 Web Components Version

2.14.X

### Browser

Chrome, Edge, Firefox, Safari

### Organization

SAP SuccessFactors

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by duygu-rmdn** [comment]
Hi @UI5/ui5-team-ctr-pir, could you please take a look?

**Comment by NakataCode** [comment]
Hi @GongRichard,

Since the `ui5-dynamic-page-header` component has been implemented according to the current accessibility specification, your request to modify the role attribute is considered a feature request. Based on the updated design requirements, this change will need to go through the InnerSource process. Could you please create a Jira ticket for Central Design and include the reference here?

Thanks and Regards,
Nikola

**Comment by GongRichard** [comment]
Hi @NakataCode ,
If you follow the accessibility specification, you don't need to remove the role="region", but you need define aria-label for it. I think it's mandatory for region role. 
You can check page https://sapui5.hana.ondemand.com/#/entity/sap.f.DynamicPage/sample/sap.f.sample.DynamicPageFreeStyle, there defined "Expanded header" aria-label  for region.

<img width="1593" height="827" alt="Image" src="https://github.com/user-attachments/assets/33d86316-a5b1-4430-aa2d-6d540b6b202b" />

Thanks,
Richard
